README.md

```diff
@@ -1,65 +1,128 @@
+maven and a full JDK to compile.
 
 ### To run the application manually:
 
-1. Compile the jar file: `mvn clean package`
-2. Create a configuration file by following the instruction in the configuration section below.
+1. Compile the jar file: `mvn clean package [-DskipITs=true]`
 
-If the manual compilation was successful, you should see a JAR file in the `target` directory, with a name like `data-exporter-VERSION_NUMBER-jar-with-dependencies.jar`. This is the file you will run with a command like the following:
+The option "-DskipITs=true" is optional and allows for running only of unit tests and not integration tests, if 
+desired, since integration tests require actual connection parameters be present in a test configuration file added
+by the user at the file path "src/test/resources/real_config.properties" (see [Integration Tests](#integration-tests)
+section for more information).
 
-4. `java -jar target/data-exporter-1.0-SNAPSHOT-jar-with-dependencies.jar [path_to_config_file]`
+If the manual compilation was successful, you should see a JAR file in the `target` directory, with a name like 
+`data-exporter-VERSION_NUMBER-jar-with-dependencies.jar`. This is the file you will run with a command like the 
+following (the `-c|--overwrite_config_file flag is optional and forces a new configuration file to be created through
+prompts from the user for new values).
 
-## Configuration
+2. `java -jar target/data-exporter-1.0-SNAPSHOT-jar-with-dependencies.jar [-c|--overwrite_config_file]`
 
-**NOTE: Configuration is done when calling the main script `runDataExporter.sh` for the first time, so this section is if set values need to be changed and for general reference.**
+## Configuration
 
-A auto-configuration script is provided at the root directory of this project and is called `configureDataExporter.sh`.  On first running the Icon Finder tool the configuration script will be run automatically and create a configuration file.   If the configuration needs to be changed, however, this script can be run directly: `./configureDataExporter.sh`.
+**NOTE: Configuration is done when calling the main script `runDataExporter.sh` for the first time (via invocation of
+the `data-exporter.jar`), so this section is if set values need to be changed as well as for general reference.**
 
-The configuration file produced by the script will be at the root directory of this project and named `config.properties`.  It can be viewed and edited directly if desired.
+The configuration file produced will be at the root directory of this project and named `config.properties`.  It can 
+be viewed and edited directly if desired.
 
-A sample configuration file is provided at `src/main/resources/config.properties` and looks like this, but should **NEVER BE EDITED DIRECTLY** and any changes are ignored by git after running the configuration script.
+A sample configuration file is provided at `src/main/resources/config.properties` and looks like this, but should 
```

Comment:
Should this be added to the .gitignore? Since its contents are unlikely to change and in case someone accidentally tries to commit something with sensitive credentials? I think I suggested something like this for one of the R packages and got denied, but I forget why

Reply:
In this case, because `sample_config.properties` (and I just realized the name is incorrect in the README and needs to be updated) is a file that is tracked by Git and present in GitHub, adding it to .gitignore isn't possible for ignoring further changes since .gitignore will only ignore untracked files (as far as I know).

The use of `git update-index -assume-unchanged ...` is what I was using to ignore any changes to the file and prevent commiting real, private configuration values to the repository.  I moved this command to the Java code, but from discussion, it's been decided to move it back to the shell script where it was before this PR

I've also been using the `git-secrets` respository locally as an extra precaution, which prevents a list of patterns you can specify from being added to any commit in a repository which `git-secrets` has been installed/enabled.